### PR TITLE
[IMP] core: remove usages of psycopg2.sql

### PR DESCRIPTION
**Pull request by kmagusiak**
Use `odoo.tools.SQL` wrapper instead.
This PR contains modifications in core and base.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by kmagusiak** [comment]
Related PRs: #173928, #173992

File: odoo/addons/base/models/ir_model.py [hunk 1]
```diff
@@ -151,12 +140,23 @@ def jsonify(val):
...
-    return [row[0] for row in model.env.cr.fetchall()]
+    comma = SQL(", ").join
+    query = SQL("""
+        INSERT INTO %(table)s (%(cols)s) VALUES %(values)s
+        ON CONFLICT (%(conflict)s) DO UPDATE SET (%(cols)s) = (%(excluded)s)
+        RETURNING id
+        """,
+        table=SQL.identifier(model._table),
+        cols=comma(SQL.identifier(fname) for fname in fnames),
+        values=comma(SQL("%s", value) for value in values),
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_model.py:155]: `SQL.join()` can actually take parameters
```suggestion
        values=comma(values),
```

File: odoo/addons/base/models/ir_sequence.py [hunk 2]
```diff
@@ -65,18 +63,15 @@ def _predict_nextval(self, seq_id):
...
+            (SELECT increment_by FROM pg_sequences WHERE sequencename = %s),
+            is_called
+        FROM %s""", seqname, seqtable)
     if self.env.cr._cnx.server_version < 100000:
-        query = sql.SQL("SELECT last_value, increment_by, is_called FROM {}")
-        params = []
-    self.env.cr.execute(query.format(seqtable), params)
+        query = SQL("SELECT last_value, increment_by, is_called FROM %s", seqtable)
+    self.env.cr.execute(query)
     (last_value, increment_by, is_called) = self.env.cr.fetchone()
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_sequence.py:75]: You may combine `execute()` and `fetchone()` as `self.env.execute_query()`.

**Review by rco-odoo — APPROVED** [review]

**Review by rco-odoo — APPROVED** [review]
@robodoo r+

**Comment by seb-odoo** [comment]
Hello, is it on purpose this prevents creating database with dot in their name?
`AssertionError: 'test.1' invalid for SQL.identifier()` at `_create_empty_database`

**Comment by kmagusiak** [comment]
> Hello, is it on purpose this prevents creating database with dot in their name?
> 
> `AssertionError: 'test.1' invalid for SQL.identifier()` in `_create_empty_database`

Hello,
It was not done on purpose, but...

It is generally a bad practice to have dots in SQL object names. Technically, you can create a database with a dot in the name just as you can create other objects if you quote them: `create table "x.y" ( "y.z" int);`. So querying in this case `"test.1".public."x.y"` becomes hard to read.

I would say the question is more whether we want to supported quoted names in `SQL.identifier`; @rco-odoo ?

**Comment by seb-odoo** [comment]
This looks like a very opinionated decision, if we want to framework to be generic enough we probably shouldn't prevent that.

This might especially be a pain point at upgrade (if we need to force people to change their DB name??) or for large custo having such identifiers in code base?

**Comment by kmagusiak** [comment]
@seb-odoo We will add back this possibility in #178406

**Pull request closed without merging**